### PR TITLE
Fix `first` and `last` functions from the `List` module to return `null` on empty

### LINKED ISSUE #311: `List.last` has design issue
Traps on empty list although it returns ?T with trap "array out of bounds"
Having `null` instead of trap would be more convenient.
Also, the semantics seems currently different to `List.first`.
Another detail: The documentation could specify the empty case for `List.first`.

### PR DESCRIPTION
**Pull request by Kamirus**
- Make `first` and `last` return `null` on empty
- Expand on docs examples with similar functions returning nulls
- Cover in tests empty and emptied lists

Closes #311

### CODE REVIEW

File: test/List.test.mo [hunk 1]
```diff
@@ -1220,4 +1276,52 @@ Test.suite(
...
+      "lastIndexOf",
+      func() {
+        Test.expect.bool(List.lastIndexOf(empty, Nat.equal, 0) == null).equal(true);
+        Test.expect.bool(List.lastIndexOf(emptied, Nat.equal, 0) == null).equal(true)
+      }
+    );
+    Test.test(
+      "firstIndexWhere",
+      func() {
+        Test.expect.bool(List.firstIndexWhere<Nat>(empty, func(x) = x == 0) == null).equal(true);
```
Comment (ggreif) [inline test/List.test.mo:1301]: Why not
```suggestion
        Test.expect.bool(List.firstIndexWhere<Nat>(empty, func x = x == 0) == null).equal(true);
```
as well as below...

**Review by rvanasa — APPROVED** [review]
LGTM, thanks!
